server/pom.xml

```diff
@@ -42,6 +42,11 @@
             <artifactId>druid-aws-common</artifactId>
             <version>${project.parent.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.druid</groupId>
+            <artifactId>druid-gcp-common</artifactId>
```

Comment:
I wish we didn't have to include gcp-common in the standard lib directory, even for people that don't use GCP. It feels like continuing the sins of our forebears (aws-common, that is). Which / how many jars does this end up adding? Trying to convince myself to be okay with this and not fight it :)

I think you said you tried doing a transitive dependency but had issues with it. For sake of getting it written down - what'd you try & what went wrong?

Reply:
I'll get a list of dependencies via the maven dependency tree here.

As far as transitive dependency, this goes back to "an extension can't depend on another extension" problem, where the implementation in an extension classloader is not accessible from any other extension. So if I put the AWS injectables in an extension, there is no other extension that can use those. All the overrides must be in the same extension.

As written elsewhere, the only place this does not apply is for things used in the hadoop world, where everything is mangled onto the same ClassLoader (often leading to all kinds of version problems). Since things in the hadoop world get mangled all together, they all can reach each other in the same classloader.

The only "real" solution I can think of for this is to have an Extension Class Loader, whereby an extension can declare a class as "exported" and have that classloader make sure any other extensions can use that common exported class from a single classloader.

Reply:
Would it work to add `aws-common` and `gcp-common` as dependencies for each extension individually, not as their own extensions? So we aren't doing extension-depends-on-extension, but are instead loading the aws/gcp common jars, separately, into each extension classloader?

Reply:
If they only provided libraries and no bindings it would. But with that approach it does not allow me to make a third party extension to inject new overrides to the HttpInitializer (for example).

And since things are bound guice makes the classloader isolation items weird for classes named the same thing but in different classloaders.

Reply:
if I recall it fails with a "you bound `HttpIntializer` multiple times" kind of error

Reply:
from the extension view point, each implementation of DruidModule is unique in its classloader, so its bindings fire independently.

Reply:
I see. That makes sense and is a bummer. But thanks for explaining it. I also guess that doing it the way I suggested would mean every extension gets its own GCP client, which is not ideal. OK, I give up, let's do it this way and hopefully someone has a flash of inspiration around making extensions able to depend on other extensions.

I built a tarball to see what new stuff this would add to `lib/`, and got:

```
./lib/checker-qual-2.5.7.jar
./lib/druid-gcp-common-0.15.0-incubating-SNAPSHOT.jar
./lib/google-api-client-1.22.0.jar
./lib/google-http-client-1.22.0.jar
./lib/google-http-client-jackson2-1.22.0.jar
./lib/google-oauth-client-1.22.0.jar
./lib/jackson-module-guice-2.6.7.jar
```

Not too bad! I was afraid it would pull in a bunch of extra deps, but, it doesn't.

Reply:
I'm super careful dude :)

Reply:
I'm just paranoid - thanks for indulging :)